code/controllers/subsystem/gravity.dm

```diff
@@ -0,0 +1,92 @@
+var/datum/subsystem/gravity/SSgravity
+var/global/legacy_gravity = FALSE
+
+/datum/subsystem/gravity
+	name = "Gravity"
+	priority = 75
+	wait = 2
+	init_order = -100
+	flags = SS_KEEP_TIMING | SS_BACKGROUND
+
+	var/list/currentrun = list()
+	var/list/currentrun_manual = list()
+	var/list/areas = list()
+	var/recalculation_cost = 0
+	var/purge_interval = -1
+	var/purge_tick = 0
+	var/purging = FALSE
+	var/list/purging_atoms = list()
+
+/datum/subsystem/gravity/New()
+	NEW_SS_GLOBAL(SSgravity)
+
+/datum/subsystem/gravity/Initialize()
+	for(var/area/A in world)
+		areas += A
+	. = ..()
+
+/datum/subsystem/gravity/proc/recalculate_atoms()
+	currentrun = list()
+	currentrun_manual = list()
+	var/tempcost = world.timeofday
```

Comment:
cost should be done with timeofday. in fact it should be done using the realtimeofday macro i made